### PR TITLE
DX-14: Referencing future changes to V2M base endpoint

### PR DESCRIPTION
**Pull request by jmulford-bandwidth**
## For the Committer

- [ ] I made sure that the site looks great and functions perfectly live
- [ ] I ran splell check
- [ ] I'm ready for these changes to be made live

### General Review

- [ ] All changes/updates requested were intentionally changed or added (no extraneous file or partial updates)
- [ ] The live rendered page behaves and looks like intended.
- [ ] All links work as intended. (click every link to make sure that it takes the user to the expected place).
- [ ] The markdown was rendered to HTML as intended. (Tables look right, code samples are gated properly).
- [ ] Any style or CSS changes are tested on multiple different pages to ensure nothing unexpected broke or changed.
- [ ] I proof-read the live site (`gitbook serve`) and there are no oblivious splelling or grammer misteaks.

### Gitbook Specific

- [ ] Any new documentation pages have been added to the `SUMMARY.md` file so they are rendered to HTML.
- [ ] Any links to other pages within this documentation use relative links (`[read more about ...](guides/voice/voicemail.md)`).
- [ ] Gitbook can install and build the documentation (`gitbook install` & `gitbook build`)

### Code Specific

- [ ] All code has been tested against the live API.
- [ ] All code can be run without throwing errors or **new** warnings.
- [ ] All code is formatted correctly and consistently (spaces, tabs).
- [ ] All code is legible and idiomatic to the language.
- [ ] All code uses sensical method, function, and variable names.
- [ ] All code samples use the appropriate syntax highlighting.
- [ ] Any examples re-use same phone numbers, uuids, variable names, etc... throughout (IE don't have something like `{from: '+19191231234', to: '+19191231234'}` unless it's meant to show the same phone number calling/texting itself ).

## TL;DR

- [ ] I did these things
- [ ] I'm ready for these changes to be made live

### CODE REVIEW

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

File: events/incomingGroup.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ For sending and receiving Group Messages, there is a maximum of 10 participants
...
-| message.media         | `array`  | A list of URLs to include as media attachments as part of the message. <br> Using the [v1 media api](http://dev.bandwidth.com/ap-docs/methods/media/getMediaMediaName.html) You can download the media **WITHIN 30 DAYS** <br><br> The URL will look something like: <br> `https://api.catapult.inetwork.com/v1/users/{userId}/media/{messageId}/{index}/{filename}` <br><br> Where `messageId` is the ID of the incoming message that had the media, `index` is the index of that media in the message (starting from `0`) and `filename` is the original filename |
+| message.media         | `array`  | A list of URLs to include as media attachments as part of the message. <br> Using the [v1 media api](http://dev.bandwidth.com/ap-docs/methods/media/getMediaMediaName.html) You can download the media **WITHIN 30 DAYS** <br><br> The URL will look something like: <br> `https://messaging.bandwidth.com/api/v2/users/{userId}/media/{messageId}/{index}/{filename}` <br><br> Where `messageId` is the ID of the incoming message that had the media, `index` is the index of that media in the message (starting from `0`) and `filename` is the original filename |
```
Comment (jdhudson3) [inline events/incomingGroup.md]: `[v2 media api]`

Will also be shortened to 2 days from 30.

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

File: events/incomingSingle.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ In order to receive text messages events, you need to ensure you have set up you
...
-| message.media         | `array`  | A list of URLs to include as media attachments as part of the message. <br> Using the [v1 media api](http://dev.bandwidth.com/ap-docs/methods/media/getMediaMediaName.html) you can download the media **WITHIN 30 DAYS** <br><br> The URL will look something like: <br> `https://api.catapult.inetwork.com/v1/users/{userId}/media/{messageId}/{index}/{filename}` <br><br> Where `messageId` is the ID of the incoming message that had the media, `index` is the index of that media in the message (starting from `0`) and `filename` is the original filename |
+| message.media         | `array`  | A list of URLs to include as media attachments as part of the message. <br> Using the [v1 media api](http://dev.bandwidth.com/ap-docs/methods/media/getMediaMediaName.html) you can download the media **WITHIN 30 DAYS** <br><br> The URL will look something like: <br> `https://messaging.bandwidth.com/api/v2/users/{userId}/media/{messageId}/{index}/{filename}` <br><br> Where `messageId` is the ID of the incoming message that had the media, `index` is the index of that media in the message (starting from `0`) and `filename` is the original filename |
```
Comment (jdhudson3) [inline events/incomingSingle.md]: `[v2 media api]`

Will also be shortened to 2 days from 30.

File: media/about.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+
+# Media
+The Media resource lets you upload your media files to Bandwidth API servers so they can be used in applications without requiring a separate hosting provider. You can upload files up to `65MB` and file storage is free for an unlimited number of files. Files are stored for up to 48 hours. Files you upload can only be accessed by you when you supply your API access `token` and `secret`. They are not available to anonymous users. Bandwidth API supports the `Cache-Control` header when you upload files.
+
+{% raw %}
+<aside class="alert general small">
+<p>
+You can upload files up to `65MB` and file storage is free for an unlimited number of files stored up to 48 hours.
```
Comment (jdhudson3) [inline media/about.md]: I _think_ we are adding a size restriction onto this (probably would be 5mb) @camdenlink1  might know more.

File: events/incomingSingle.md (hunk 2 above)
Comment (jmulford-bandwidth) [reply events/incomingSingle.md]: 👍 Should we also change the v1 verbiage to be v2 as well?

File: messaging2Migration.md [hunk 4]
```diff
@@ -34,7 +34,7 @@ Authorization: {apiToken:apiSecret}
     "text" : "Hello there from Bandwidth!"
 }
 {%- language name="v2/messages", type="http" -%}
-POST https://api.catapult.inetwork.com/v2/users/{userId}/messages HTTP/1.1
+POST https://messaging.bandwidth.com/v2/users/{userId}/messages HTTP/1.1
```
Comment (jdhudson3) [inline messaging2Migration.md]: `/api/v2`

File: events/incomingSingle.md (hunk 2 above)
Comment (jdhudson3) [reply events/incomingSingle.md]: Probably, can you give me an example?

Comment (jmulford-bandwidth) [reply events/incomingSingle.md]: Actually will this endpoint even exist in V2 media? The docs you wrote did not include the `/media/{messageId}/{index}/{filename}` endpoint

Comment (jdhudson3) [reply events/incomingSingle.md]: Both `/media/{uploaded-file-name}` and `/media/{messageId}/{index}/{filename}` reference the same endpoint.

File: messaging2Migration.md (hunk 4 above)
Comment (jmulford-bandwidth) [reply messaging2Migration.md]: 👍 fixed

File: media/about.md (hunk 3 above)
Comment (jmulford-bandwidth) [reply media/about.md]: Discussion outside of here lead to the conclusion

* Keep this as 65mb limit
* Add 5mb limit for media included on messages

File: events/incomingSingle.md (hunk 2 above)
Comment (jmulford-bandwidth) [reply events/incomingSingle.md]: Gotcha

Comment (jmulford-bandwidth) [reply events/incomingSingle.md]: To be consistent with the V2M docs, I changed this part to use the `/media/{uploaded-file-name}`

File: events/incomingGroup.md (hunk 1 above)
Comment (jmulford-bandwidth) [reply events/incomingGroup.md]: fixed

File: events/incomingSingle.md (hunk 2 above)
Comment (jmulford-bandwidth) [reply events/incomingSingle.md]: fixed

File: media/about.md (hunk 3 above)
Comment (jmulford-bandwidth) [reply media/about.md]: fixed

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

File: media/about.md (hunk 3 above)
Comment (camdenlink1) [reply media/about.md]: i think we decided to go with 5 mb as the limit on media uploads. see https://bandwidth.slack.com/archives/GF7C21F6U/p1551462839002000 for confirmation from Jason Sommerset

Comment (jmulford-bandwidth) [reply media/about.md]: This comment is outdated, the current file has 5MB as the limit

https://github.com/Bandwidth/v2-messaging/blob/bf0ce32236ba19db186531440142ced2b249334d/media/about.md

File: README.md [hunk 5]
```diff
@@ -2,7 +2,9 @@
 
 **PLEASE READ**: This is the home for customers using Bandwidth's Messaging v2. If you don't know which version of the messaging API to use, or if you are using our free trial, you're probably on our Messaging v1 API [(see docs)](https://dev.bandwidth.com/ap-docs/methods/restApi.html). Interested in our Messaging v2 API? Request access [here](https://go.bandwidth.com/messaging-v2-api.html).
 
-<div class="alert alert--general"><p>Bandwidth's Messaging v2 does not store <b>ANY</b> messages or records for fetching later. If you need message history you need to implement a storage solution that meets your needs.</p></div>
+<div class="alert alert--general"><p>The base URL for the Messaging API is changing in early April 2019. The new base URL will be <code>https://messaging.bandwidth.com/api/v2</code> and is replacing the old base URL which is <code>https://api.catapult.inetwork.com/v2</code>. The old base URL will be deprecated in June 2019. Current customers will be notified when the new URL is ready to be used, and should not switch until notified.</p></div>
```
Comment (tato123) [inline README.md]: Do we know when this will be deleted? I'd like to create a task for deleting out anything time related

File: media/deleteMedia.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+{% method %}
+
+## Remove Media File
+Deletes a media file from Bandwidth API server. Make sure you don't have any application scripts still using the media before you delete. If you accidentally delete a media file, you can immediately upload a new file with the same name.
+
+### Request URL
+
+<code class="delete">DELETE</code>`https://messaging.bandwidth.com/api/v2/users/{userId}/media/{mediaName}`
+
+**Notice: This URL will be active in April 2019. Current customers will be notified once this URL is ready.**
```
Comment (tato123) [inline media/deleteMedia.md]: **Note**: Should have some sort of roll off strategy to remove this when no longer relevant / needed

File: README.md (hunk 5 above)
Comment (jmulford-bandwidth) [reply README.md]: We have a potential release date of April 4th for the new base url. I made DX-28 and DX-29 to reflect future updates to the docs

**Comment by jmulford-bandwidth** [comment]
> uisetup.md and messaging2Migration both cut over without any explanation of the April timeline.
> 
> Overall these changes seem straight forward but they just seem fragile (prone to being stale, old, outdated documentation quickly after the April timeline)

Fixed with newest commit

File: media/deleteMedia.md (hunk 6 above)
Comment (jmulford-bandwidth) [reply media/deleteMedia.md]: DX-28 is the ticket for this

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

**Comment by jmulford-bandwidth** [comment]
As a request from Carissa this is ready to merge

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/

File: media/uploadMedia.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+You can upload files up to <code>3.75MB</code> and file storage is free for an unlimited number of files within a 48 hour retention policy.
+</p>
+</aside>
+
+### Properties
+| Header         | Description                                                                                                                        | Mandatory |
+|:---------------|:-----------------------------------------------------------------------------------------------------------------------------------|:----------|
+| Content-Length | Indicates the size of the entity-body.                                                                                             | Yes       |
+| Cache-Control  | General-header field is used to specify directives that MUST be obeyed by all caching mechanisms along the request/response chain. | No        |
+| Content-Type   | The media type of the entity-body.                                                                                                 | No        |
```
Comment (camdenlink1) [inline media/uploadMedia.md]: i think that we decided that content-type should be mandatory, let me double check on that and i'll let you know

Comment (camdenlink1) [reply media/uploadMedia.md]: yeah content-type is mandatory, other than that everything looks good

Comment (jmulford-bandwidth) [reply media/uploadMedia.md]: Fixed

**Comment by DX-Bandwidth** [comment]
Preview Changes at:
        http://v2-messaging-DX-14.s3-website-us-east-1.amazonaws.com/
